server/src/main/java/org/opensearch/wlm/cancellation/DefaultTaskSelectionStrategy.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.wlm.QueryGroupTask;
+import org.opensearch.wlm.ResourceType;
+import org.opensearch.wlm.tracker.TaskResourceUsageCalculator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.opensearch.wlm.cancellation.DefaultTaskCancellation.MIN_VALUE;
+
+/**
+ * Represents an abstract task selection strategy.
+ * This class implements the DefaultTaskSelectionStrategy interface and provides a method to select tasks for cancellation based on a sorting condition.
+ * The specific sorting condition depends on the implementation.
+ */
+public class DefaultTaskSelectionStrategy {
```

Comment:
The naming seems confusing. Should we instead call it DefaultTaskCancellationStrategy or something like that? As its main intention is to select tasks for cancellation which isn't conveyed currently.

Reply:
Yes, I will rename this to `LongestRunningTaskFirstSelectionStrategy`